src/os/unix/posix/memfd.rs

```diff
@@ -32,31 +32,26 @@ pub fn memfd_open() -> Result<c_int> {
 
 #[cfg(not(any(target_os = "linux", target_os = "android", target_os = "freebsd")))]
 pub fn memfd_open() -> Result<c_int> {
-    use libc::c_char;
-    use rand::distributions::Alphanumeric;
-    use rand::{thread_rng, Rng};
-
     const OFLAGS: c_int = libc::O_RDWR | libc::O_CREAT | libc::O_EXCL | libc::O_CLOEXEC;
+    let mut path_bytes: [u8; 14] = *b"/vmap-XXXXXXX\0";
 
-    let mut path: [u8; 18] = *b"/tmp/vmap-XXXXXXX\0";
-
-    let mut rng = thread_rng();
-    let end = path.len() - 1;
-
-    loop {
-        for dst in &mut path[10..end] {
-            *dst = rng.sample(&Alphanumeric) as u8;
-        }
+    for i in (0..10000000).cycle() {
+        let path = {
+            use std::io::Write;
+            write!(&mut path_bytes[6..], "{:0>7}", i).unwrap();
```

Comment:
What if instead of an iterative cycle, we try for a less deterministic value? One thought could be to use something approximately time-based with an integer mix, for example:

```rust
fn id() -> u32 {
    let mut t = SystemTime::now()
        .duration_since(UNIX_EPOCH)
        .unwrap()
        .subsec_nanos();
    
    // Mix using Jenkins' integer hash
    t = t.wrapping_add(0x7ed55d16).wrapping_add(t << 12);
    t = (t ^ 0xc761c23c) ^ (t >> 19);
    t = t.wrapping_add(0x165667b1).wrapping_add(t << 5);
    t = t.wrapping_add(0xd3a2646c) ^ (t << 9);
    t = t.wrapping_add(0xfd7046c5).wrapping_add(t << 3);
    t = (t ^ 0xb55a4f09) ^ (t >> 16);
    t
}
```

And then we could do:

```rust
write!(&mut path_bytes[6..], "{:0>7}", id()).unwrap();
```

Reply:
Hmm... using the time works but I'm not sure using the hash is necessary.  The output space will have the same number of unique values (1e9) right?

Reply:
Yeah the output space would have exact the same number of unique values, but they would't be sequential. Though, really, that may not actually provide much utility here.

Reply:
I was hoping to avoid the extra syscall, but this is probably way to go.  Updated it to use the time directly.

Reply:
Yeah, totally do get that. I was thinking the trade-off between checking the time would be preferable to the syscall for attempting to create a path that already exists. With several running at the same time, I think there would be fewer overall calls due to the distribution of names.